src/Compilers/Core/MSBuildTask/Vbc.cs

```diff
@@ -909,7 +903,7 @@ private bool InitializeHostCompiler(IVbcHostObject vbcHostObject)
                 }
 
                 // Check for support of the LangVersion property
-                if (vbcHostObject is IVbcHostObject3)
+                if (vbcHostObject is IVbcHostObject3 && VbcHostObjectSupports(LangVersion))
```

Comment:
You might want to add an additional check that says if the host doesn't support it _and_ the host object doesn't implement ICompileOptionsHostObject then we issue an error or some fallback. Because in that case (consider Dev14 RTM) then you're just dropping language version on the floor and probably confusing something...

Not sure if that's a useful scenario though.